packages/core/src/sync/index.ts

```diff
@@ -152,26 +150,29 @@ export const createSync = async (args: CreateSyncParameters): Promise<Sync> => {
       return undefined;
     }
 
-    const checkpoints = [...localSyncs.entries()]
-      // Remove completed networks from checkpoint calculation for "latest"
-      .filter(
-        ([, localSync]) => tag !== "latest" || localSync.isComplete() === false,
-      )
-      .map(([network, localSync]) => {
-        const block = localSync[`${tag}Block`]!;
-
-        // The checkpoint returned by this function is meant to be used in
-        // a closed interval (includes endpoints), so "start" should be inclusive.
-        return blockToCheckpoint(
-          block,
-          network.chainId,
-          tag === "start" ? "down" : "up",
-        );
-      });
-
-    // Return `true` if all networks have been filtered out
+    let checkpoints = [...localSyncs.entries()];
+    if (tag === "latest")
+      checkpoints = checkpoints.filter(
+        ([, localSync]) => localSync.isComplete() === false,
+      );
+    // Return early if all networks are complete
     if (checkpoints.length === 0) return undefined;
```

Comment:
To confirm: If I ask for the "finalized" block when all historical syncs are complete, I'll get `undefined` ?